### DISCUSSION TITLE
Follow XDG Base Directory specification for clusterctl configuration file(s)

### DISCUSSION THREAD

**Discussion by pavel-github** [kubernetes-sigs/cluster-api#8832] [Ideas]
Instead of polluting user $HOME directory, it would be better UX to follow [XDG Base Directory specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) and keep configuration files under `$HOME/.config/cluster-api/*`.

Since clusterctl uses Viper to load configuration (https://github.com/kubernetes-sigs/cluster-api/blob/main/cmd/clusterctl/client/config/reader_viper.go#L63), we can add an additional configuration directory to not break an existing behavior.

#### Technical details:
- Go library for getting XDG folders: https://github.com/adrg/xdg

Let me know if the idea is worth, so I could provide a PR.

**Comment by killianmuldoon** [comment]
Generally issues are a better venue for bringing up ideas like this.

There is already a PR open for this: https://github.com/kubernetes-sigs/cluster-api/pull/6913. The issue thus far has been reviewer bandwidth more than anything.

**Reply by pavel-github** [reply]
@killianmuldoon , thanks for the answer (for some reason GitHub search didn't catch it when I was searching 😄).
I'm going to close this topic.
